lms/djangoapps/bookmarks/models.py

```diff
@@ -52,6 +61,10 @@ def create(cls, bookmarks_data, block=None):
         bookmarks_data['_path'] = json.dumps(cls.get_path(block))
 
         bookmark, __ = cls.objects.get_or_create(**bookmarks_data)
+        bookmark.block_cache = XBlockCache.objects.get_or_create(
+            course_key=bookmark.course_key, usage_key=usage_key,
+        )
+        bookmark.save()
```

Comment:
`get_or_create` returns a tuple of `(object, created)`. You don't need to save if `created` is true. Also, you should be passing in `defaults={...}` to set attributes in the case that the XBlockCache doesn't yet exist.